cloudconfig.go

```diff
@@ -0,0 +1,71 @@
+package cloudconfig
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"html/template"
+	"io/ioutil"
+
+	"github.com/giantswarm/awstpr"
+	tpraws "github.com/giantswarm/awstpr/aws"
+	"github.com/giantswarm/clustertpr/node"
+)
+
+type AWSNode struct {
+	Node    node.Node
+	AwsInfo tpraws.Node
+}
+
+type CloudConfigTemplateParams struct {
+	Spec      awstpr.Spec
+	Node      AWSNode
+	TLSAssets CompactTLSAssets
+}
+
+type CloudConfig struct {
+	config   string
+	params   CloudConfigTemplateParams
+	path     string
+	template string
+}
+
+func NewCloudConfig(templatePath string, params CloudConfigTemplateParams) (*CloudConfig, error) {
+	cloudConfigTempl, err := ioutil.ReadFile(templatePath)
```

Comment:
However, while I have nothing against writing mock cloudconfigs to test, I think we should still test also the actual cloudconfigs which we have in this repo. That would be very helpful for checking whether the changes in cloudconfig files are correct.